storage/database/dynamodb_test.go

```diff
@@ -85,6 +90,68 @@ func (s *SuiteDynamoDB) TestDynamoDB_Put() {
 	s.NoError(returnedErr)
 }
 
+// TestDynamoDB_Timeout tests if a timeout error occurs.
+// When there is no answer from DynamoDB server due to network failure,
+// a timeout error should occur.
+// A fake server is setup to simulate a server with a response latency.
+func (s *SuiteDynamoDB) TestDynamoDB_Timeout() {
+	// fakeEndpoint allows TCP handshakes, but doesn't answer anything to client.
+	// The fake server is used to produce a network failure scenario.
+	fakeEndpoint := "127.0.0.1:14566"
+	maxRetries := 2
+	timeout := 1 * time.Second
+	go func() {
+		tcpAddr, err := net.ResolveTCPAddr("tcp", fakeEndpoint)
+		if err != nil {
+			s.FailNow(err.Error())
+		}
+
+		listen, err := net.ListenTCP("tcp", tcpAddr)
+		if err != nil {
+			s.FailNow(err.Error())
+		}
+		defer listen.Close()
+
+		for {
+			// Deadline prevents infinite waiting of the fake server
```

Comment:
Just one more comment :)
```suggestion
			// Deadline prevents infinite waiting of the fake server
			// Wait longer than (maxRetries+1) * timeout
```